src/approaches/interactive_learning_approach.py

```diff
@@ -255,6 +258,75 @@ def _create_glib_interaction_strategy(
         termination_function = task.goal_holds
         return act_policy, termination_function
 
+    def _create_greedy_lookahead_interaction_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Sample a certain number of max-length trajectories and pick the one
+        that has the highest cumulative score."""
+        init = self._train_tasks[train_task_idx].init
+        # Sample trajectories by sampling random sequences of NSRTs.
+        best_score = -np.inf
+        best_options = []
+        for _ in range(CFG.interactive_max_num_trajectories):
+            state = init.copy()
+            options = []
+            trajectory_length = 0
+            total_score = 0.0
+            while trajectory_length < CFG.interactive_max_trajectory_length:
+                # Sample an NSRT that has preconditions satisfied in the
+                # current state.
+                ground_nsrt = self._sample_applicable_ground_nsrt(state)
+                if ground_nsrt is None:  # No applicable NSRTs
+                    break
+                assert all(a.holds for a in ground_nsrt.preconditions)
+                # Sample an option. Note that goal is assumed not used.
+                option = ground_nsrt.sample_option(state,
+                                                   goal=set(),
+                                                   rng=self._rng)
+                # Assume for now that options will be initiable when the
+                # preconditions of the NSRT are satisfied.
+                assert option.initiable(state)
+                state, num_actions = \
+                    self._option_model.get_next_state_and_num_actions(state,
+                                                                      option)
+                # Special case: if the num actions is 0, something went wrong,
+                # and we don't want to use this option after all. To prevent
+                # possible infinite loops, just break immediately in this case.
+                if num_actions == 0:
+                    break
+                options.append(option)
+                trajectory_length += num_actions
+                # Update the total score.
+                atoms = utils.abstract(state, self._predicates_to_learn)
+                total_score += self._score_atom_set(atoms, state)
+            else:  # Sampled a complete trajectory
```

Comment:
is there a reason you want to force all the trajectories to be exactly of length `interactive_max_trajectory_length`? what's the problem if they're shorter than that?

Reply:
i was thinking about this too... i guess it's not a problem and if it's the best trajectory then we should use it!